### ISSUE TITLE
Having more than one authentication configurations for a request, only evaluate the first

### ISSUE THREAD

**Issue by gdesantos** [ktorio/ktor#1586]
**Ktor Version and Engine Used (client or server and name)**
ktor server version 1.3.0

**Describe the bug**
Having a simple application, with 3 JWT authentication providers installed, if i write a request authenthicated with the 3 providers, only the first is evaluated.

**To Reproduce**
Here is a simple full code to reproduce it

```
package com.example

import com.auth0.jwt.JWT
import com.auth0.jwt.algorithms.Algorithm
import io.ktor.application.*
import io.ktor.response.*
import io.ktor.request.*
import io.ktor.routing.*
import io.ktor.http.*
import io.ktor.auth.*
import com.fasterxml.jackson.databind.*
import io.ktor.auth.jwt.jwt
import io.ktor.jackson.*
import io.ktor.features.*

fun main(args: Array<String>): Unit = io.ktor.server.netty.EngineMain.main(args)

@Suppress("unused") // Referenced in application.conf
@kotlin.jvm.JvmOverloads
fun Application.module(testing: Boolean = false) {

    install(ContentNegotiation) {
        jackson {

        }
    }

    install(Authentication) {
        val jwtVerifier = JWT
            .require(Algorithm.HMAC512("secret"))
            .withIssuer("issuer")
            .build()

        jwt("auth1") {
            verifier(jwtVerifier)
            validate { credential ->
                print("Validating 1")
                null
            }
        }

        jwt("auth2") {
            verifier(jwtVerifier)
            validate { credential ->
                print("Validating 2")
                null
            }
        }

        jwt("auth3") {
            verifier(jwtVerifier)
            validate { credential ->
                print("Validating 3")
                null
            }
        }
    }

    routing {
        post("/auth") {
            val token = JWT.create()
                .withIssuer("issuer")
                .sign(Algorithm.HMAC512("secret"))
            call.respond(hashMapOf("token" to token))
        }

        authenticate("auth1", "auth2", "auth3") {
            get("test") {
                call.respond(HttpStatusCode.OK)
            }
        }
    }
}
```

To reproduce the bug, first call the /auth POST to obtain a token, and then call /test GET with that token trying to check the token against the 3 providers.
 
**Expected behavior**
The log should write:
Validating 1
Validating 2
Validating 3

But the log only write:
Validating 1

This means that only the first JWT authentication provider is being evaluated, but not the other two. The first JWT authentication provider is returning null, so ktor engine should try with the next

This a short example of what i am having in a real environment with a much more complex code, writing this example to focus only on this problem.

**Comment by gdesantos** [comment]
Doing more testing:

It seems that if i throw an exception instead of return null, all the authentication providers are evaluated.
Using this:

```
        jwt("auth1") {
            verifier(jwtVerifier)
            validate { credential ->
                print("Validating 1")
                throw Exception("TEST1")
            }
        }

        jwt("auth2") {
            verifier(jwtVerifier)
            validate { credential ->
                print("Validating 2")
                throw Exception("TEST1")
            }
        }

        jwt("auth3") {
            verifier(jwtVerifier)
            validate { credential ->
                print("Validating 3")
                throw Exception("TEST3")
            }
        }
```

But the documentation for the validate function:

```
    /**
     * Apply [validate] function to every call with [JWTCredential]
     * @return a principal (usually an instance of [JWTPrincipal]) or `null`
     */
```

So is the documentation wrong?

**Comment by cy6erGn0m** [comment]
Fixed in 1.3.1

**Comment by gdesantos** [comment]
Thanks and keep up the good work!
